### DISCUSSION TITLE
Upgraded from IKVM 8.5 to 8.10, and now I'm getting "An item with the same key has already been added" error.

### DISCUSSION THREAD

**Discussion by apn-hroussel** [ikvmnet/ikvm#591] [Q&A] [answered]
Hello,

I had a asp.net application, converted from Java (jars) using the ikvmc compiler.  Everything was working fine under IKVM 8.5.

After the update to IKVM 8.10.1 I was having issues with ikvm.home not found (despite having the same ikvm.properties file in my application's bin directory).  I tried the new app.config directive, and so by adding <add key="ikvm:ikvm.home" value="ikvm" /> to my web.config that error was gone, but now I'm getting,

"An item with the same key has already been added"

Any ideas on how to fix this?

Thanks

**Comment by wasabii** [comment]
Stack trace or anything?

**Reply by apn-hroussel** [reply]
Yes here it is,

[ArgumentException: An item with the same key has already been added.]
   System.ThrowHelper.ThrowArgumentException(ExceptionResource resource) +60
   System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add) +14653218
   IKVM.Runtime.Properties.InitSystemProperties(Dictionary`2 p) +1232
   IKVM.Runtime.Properties.GetInitProperties() +44
   System.Lazy`1.CreateValue() +732
   System.Lazy`1.LazyInitValue() +189
   IKVM.Runtime.LibJvm.JVM_InitProperties(JNIEnv* env, IntPtr props) +68

[InternalError: cli.System.ArgumentException: An item with the same key has already been added.]
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +32
   IKVM.Runtime.JVM.ThrowPendingException() +82
   IKVM.Runtime.JNI.JNIFrame.Leave() +53
   java.lang.System.initProperties(Properties props) +543
   java.lang.System.initializeSystemClass() +115
   IKVM.Runtime.JVM.Init() +456
   IKVM.Runtime.RuntimeInit.Init() +100
   .cctor() +15

[TypeInitializationException: The type initializer for '<Module>' threw an exception.]
   ikvm.runtime.Startup.init() +0
   .cctor() +29

**Reply by wasabii** [reply, marked as answer]
This should be resolved in 8.10.2

**Reply by apn-hroussel** [reply]
Thanks, that fixed the issue!

**Comment by djenoufrank** [comment]
@apn-hroussel Hello!
For this problem you said :
 "After the update to IKVM 8.10.1 I was having issues with ikvm.home not found (despite having the same ikvm.properties file in my application's bin directory). I tried the new app.config directive, and so by adding to my web.config that error was gone."
 
 
 Please just want to know what have you add in your web.config to fix ikvm.home not found problem ?

**Reply by apn-hroussel** [reply]
It's a while since I did these tests, but in my web.config I had the following key,

&lt;add key="ikvm:ikvm.home" value="ikvm" /&gt;

**Reply by djenoufrank** [reply]
Thanks for your reply.
I try it but I have the same problem. I change relative path ikvm for absolute one but nothing have change.
